R/validate_target_data.R

```diff
@@ -0,0 +1,186 @@
+#'   file_path = "oracle-output/output_type=pmf/part-0.parquet",
+#'   target_type = "oracle-output"
+#' )
+#' @export
+validate_target_data <- function(
+  hub_path,
+  file_path,
+  target_type = c("time-series", "oracle-output"),
+  date_col = NULL,
+  na = c("NA", ""),
+  output_type_id_datatype = c(
+    "from_config",
+    "auto",
+    "character",
+    "double",
+    "integer",
+    "logical",
+    "Date"
+  ),
+  validations_cfg_path = NULL,
+  round_id = "default"
+) {
+  target_type <- rlang::arg_match(target_type)
+  output_type_id_datatype <- rlang::arg_match(output_type_id_datatype)
+
+  checks <- new_hub_validations()
+
+  # -- File parsing checks ----
+  checks$target_file_read <- try_check(
+    check_target_file_read(
+      file_path = file_path,
+      hub_path = hub_path
+    ),
+    file_path
+  )
+  if (is_any_error(checks$target_file_read)) {
+    return(checks)
+  }
+
+  # If `csv` file, read in using hub schema. Otherwise use file
+  # schema for other file formats.
+  if (fs::path_ext(file_path) == "csv") {
+    target_tbl <- read_target_file(
+      target_file_path = file_path,
+      hub_path = hub_path,
+      coerce_types = "target"
+    )
+  } else {
+    target_tbl <- read_target_file(
+      target_file_path = file_path,
+      hub_path = hub_path,
+      coerce_types = "none"
+    )
+  }
+
+  # -- Standard checks ----
+  checks$target_tbl_colnames <- try_check(
+    check_target_tbl_colnames(
+      target_tbl = target_tbl,
+      target_type = target_type,
+      file_path = file_path,
+      hub_path = hub_path
+    ),
+    file_path
+  )
+  if (is_any_error(checks$target_tbl_colnames)) {
+    return(checks)
+  }
+  checks$target_tbl_coltypes <- try_check(
+    check_target_tbl_coltypes(
+      target_tbl = target_tbl,
+      target_type = target_type,
+      date_col = date_col,
+      na = na,
+      output_type_id_datatype = output_type_id_datatype,
+      file_path = file_path,
+      hub_path = hub_path
+    ),
+    file_path
+  )
+  if (is_any_error(checks$target_tbl_coltypes)) {
+    return(checks)
+  }
+  checks$target_tbl_ts_targets <- try_check(
+    check_target_tbl_ts_targets(
+      target_tbl = target_tbl,
+      target_type = target_type,
+      file_path = file_path,
+      hub_path = hub_path
+    ),
+    file_path
+  )
+  if (is_any_error(checks$target_tbl_ts_targets)) {
+    return(checks)
+  }
+
+  checks$target_tbl_rows_unique <- try_check(
+    check_target_tbl_rows_unique(
+      target_tbl = target_tbl,
+      target_type = target_type,
+      file_path = file_path,
+      hub_path = hub_path
+    ),
+    file_path
+  )
+
+  target_tbl_chr <- hubData::coerce_to_character(target_tbl)
+
+  checks$target_tbl_values <- try_check(
+    check_target_tbl_values(
+      target_tbl_chr = target_tbl_chr,
+      target_type = target_type,
+      file_path = file_path,
+      hub_path = hub_path
+    ),
+    file_path
+  )
+  if (is_any_error(checks$target_tbl_values)) {
+    return(checks)
+  }
```

Comment:
Same as my other question, Can I ask why this one stop the checking and returns an error? Will it cause issue in the next checks if it fails?

Reply:
This one affects the output type id check. The task ID values need to be correct because the output type IDs are checked for validity and completeness by grouping by task IDs. So if task IDs contain invalid values, we can't guarantee the output type id check result is valid.